### PR TITLE
add ability to ship remote only branch

### LINKED ISSUE #593: git ship: remote branch
Use case: 
Create a branch on one machine. PR gets approved and try to ship it from another machine.

```
git ship <remote-branch-name>
```
outputs
```
  Error
  There is no branch named '<remote-branch-name>'
```

git ship should do a fetch first in order to allow remote only branches to be shipped.

### PR DESCRIPTION
**Pull request by charlierudolph**
@kevgo @allewun @ricmatsui 

fixes #593

### CODE REVIEW

File: features/git-ship/supplied_branch/feature_branch/remote_only_branch.feature [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+      |               | git checkout main                  |
+      | main          | git merge --squash feature         |
+      |               | git commit -m "feature done"       |
+      |               | git push                           |
+      |               | git push origin :feature           |
+      |               | git branch -D feature              |
+      |               | git checkout other-feature         |
+      | other-feature | git stash pop                      |
+    And I end up on the "other-feature" branch
+    And my workspace still has an uncommitted file with name: "feature_file" and content: "conflicting content"
```
Comment (kevgo) [inline features/git-ship/supplied_branch/feature_branch/remote_only_branch.feature]: Hmm, I think we should make a step `And my workspace still has the uncommitted file` here. What do you think? Right now the definition of file name and content are redundant.

File: src/git-ship [hunk 2]
```diff
@@ -38,6 +43,9 @@ function preconditions {
   ensure_is_feature_branch "$target_branch_name" "Only feature branches can be shipped."
   ensure_knows_parent_branches "$target_branch_name"
   ensure_has_no_unshipped_ancestors "$target_branch_name"
+  if [ "$HAS_REMOTE" = true -a "$fetched" = false ]; then
```
Comment (kevgo) [inline src/git-ship]: We use two different styles for boolean and in this method now:

```bash
if [ "$#" -gt 0 ] && [[ ! "$1" =~ ^- ]]; then
```

as well as

```bash
if [ "$HAS_REMOTE" = true -a "$fetched" = false ]; then
```

Let's agree on one style and use it consistently.

**Comment by kevgo** [comment]
Nice! :ship: after the comments are addressed.

As a separate change, what do you think about making `fetch` something that is automatically done by the GT framework unless the command sets a variable to omit it? I think most commands need to fetch anyways.

File: features/git-ship/supplied_branch/feature_branch/remote_only_branch.feature (hunk 1 above)
Comment (charlierudolph) [inline features/git-ship/supplied_branch/feature_branch/remote_only_branch.feature]: Fixed. I copied this from another PR so looks like there are some files that didn't get changed over in #482

**Comment by charlierudolph** [comment]
I don't know about fetch becoming a part of the framework because each command can optimize when it should actually fetch in order to deal with checking preconditions.

File: src/git-ship (hunk 2 above)
Comment (charlierudolph) [inline src/git-ship]: Converted to `&&` and created #597 to make the rest consistent

**Comment by charlierudolph** [comment]
Also was able to simplify the code and we already have the code in place to ensure fetch only happens once per command.

File: features/git-ship/supplied_branch/feature_branch/remote_only_branch.feature (hunk 1 above)
Comment (charlierudolph) [inline features/git-ship/supplied_branch/feature_branch/remote_only_branch.feature]: Sorry my mistake there. Yes this is a possible cleanup. I'll deal with it separately

File: features/step_definitions/branch_steps.rb [hunk 3]
```diff
@@ -6,9 +6,13 @@
 end
 
 
-Given(/^I have a( local)?(?: feature)? branch named "([^"]+)"$/) do |local, branch_name|
-  create_branch branch_name, remote: !local
-  set_parent_branch branch: branch_name, parent: 'main', ancestors: 'main'
+Given(/^I have a( local)?(?: feature)? branch named "([^"]+)"( on another machine)?$/) do |local, branch_name, remote|
+  user = 'developer'
+  user += '_secondary' if remote
```
Comment (kevgo) [inline features/step_definitions/branch_steps.rb]: That's a cool trick btw!

**Comment by kevgo** [comment]
:ship:

**Pull request closed without merging**